source/processes/fs.cpp

```diff
@@ -1130,22 +1130,25 @@ class ArchiveSystemSaveDataForSaveId : public ArchiveNonPXI {
 
 class ArchiveSaveDataProvider {
```

Comment:
Just noting down one problem I found so far for reference:

`CreateFile("/../test")` is expected to return an error, but this PR will construct a sandbox-escaping path instead. This is caught by PathValidator:

> [FRONTEND] [error] Requested path leaked out of sandbox: Got path ~/.local/share/mikage/data/data/0000000000000000/extdata/00048000/f00000ee/test, expected a child of ~/.local/share/mikage/data/data/0000000000000000/extdata/00048000/f00000ee/user